### PR TITLE
feat: Provide option to connect to external mongodb (#5369)

### LINKED ISSUE #5369: Core: Update services to consider mongo connection string with replica set
## Action Item

Update all services that use mongo to use a connection string with Replica Set.

## Technical Details

The connection string should also be configurable via helm values.

More info about the syntax and format of the MongoDB connection string is available here: https://docs.mongodb.com/manual/reference/connection-string/

## DoD

- [ ] All services using MongoDB can connect via Replica Set
- [ ] The connection string can be specified via helm value
- [ ] Unit test
- [ ] Integration test can be executed locally with an external MongoDB

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #5369

This PR allows to connect Keptn to connect to any external Mongodb instance by setting the following variables to the installer helm chart:

- `control-plane.mongodb.external.connectionString` - If this value is set, keptn services using the mongodb will connect themselves to the provided URL (example: `mongodb+srv://<user>:<password>@my-db.mongodb.net/myFirstDatabase?retryWrites=true&w=majority`) 
- `control-plane.mongodb.dbName` - Specifies the name of the database. This value is necessary because the name of the database provided in the connection string is not necessarily the same than the actual database to be used (e.g. the connection string could refer to a separate authentication database)

### CODE REVIEW

File: installer/manifests/keptn/charts/control-plane/charts/mongodb/templates/deployment.yaml [hunk 1]
```diff
@@ -75,3 +76,4 @@ spec:
         - name: mongodata
           persistentVolumeClaim:
             claimName: mongodata
+{{- end -}}
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/charts/mongodb/templates/deployment.yaml:79]: please add the missing newline :)

File: installer/manifests/keptn/charts/control-plane/charts/mongodb/templates/pvc.yaml [hunk 2]
```diff
@@ -17,4 +18,5 @@ spec:
       storage:  {{ .Values.storage }}
   {{- if .Values.storageClass }}
   storageClassName: {{ .Values.storageClass }}
-  {{- end }}
\ No newline at end of file
+  {{- end }}
+{{- end -}}
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/charts/mongodb/templates/pvc.yaml:22]: please add the missing newline :)

File: installer/manifests/keptn/charts/control-plane/templates/core.yaml [hunk 3]
```diff
@@ -356,7 +356,13 @@ spec:
                   name: mongodb-credentials
                   key: password
             - name: MONGO_DB_NAME
-              value: 'keptn'
+              value: {{ .Values.mongodb.dbName | default "keptn"}}
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/templates/core.yaml:359]: Do you really need the default here, if the name is already provided in the values file directly?

File: installer/manifests/keptn/charts/control-plane/templates/core.yaml [hunk 4]
```diff
@@ -758,7 +764,13 @@ spec:
                   name: mongodb-credentials
                   key: password
             - name: MONGO_DB_NAME
-              value: keptn
+              value: {{ .Values.mongodb.dbName | default "keptn"}}
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/templates/core.yaml:767]: Do you really need the default here, if the name is already provided in the values file directly?

File: installer/manifests/keptn/charts/control-plane/templates/mongodb-credentials.yaml [hunk 5]
```diff
@@ -27,6 +32,10 @@
 {{- $mongoAdminPassword = index $mongosecret.data "admin_password" -}}
 {{- end -}}
 
+{{- if $mongosecret -}}
+{{- $mongoExternalConnectionString = index $mongosecret.data "external_connection_string" -}}
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/templates/mongodb-credentials.yaml:36]: I would add a check here to see if the value is really set in the secret, before referencing it. Somthing like `{{ if index $mongosecret.data "external_connection_string" }}`

File: installer/manifests/keptn/charts/control-plane/templates/mongodb-datastore.yaml [hunk 6]
```diff
@@ -63,8 +63,8 @@ spec:
           value: "{{ .Values.prefixPath }}"
         - name: MONGODB_HOST
           value: '{{ .Values.mongodb.host }}'
-        - name: MONGODB_DATABASE
-          value: 'keptn'
+        - name: MONGO_DB_NAME
+          value: {{ .Values.mongodb.dbName | default "keptn"}}
```
Comment (mowies) [inline installer/manifests/keptn/charts/control-plane/templates/mongodb-datastore.yaml:67]: Do you really need the default here, if the name is already provided in the values file directly?

**Review by mowies — CHANGES_REQUESTED** [review]
Left some comments :)

File: installer/manifests/keptn/charts/control-plane/templates/mongodb-datastore.yaml (hunk 6 above)
Comment (bacherfl) [reply installer/manifests/keptn/charts/control-plane/templates/mongodb-datastore.yaml:67]: the name should be provided in the values.yaml file, I just added the default to be safe

File: installer/manifests/keptn/charts/control-plane/templates/core.yaml (hunk 4 above)
Comment (bacherfl) [reply installer/manifests/keptn/charts/control-plane/templates/core.yaml:767]: the name should be provided in the values.yaml file, I just added the default to be safe

File: installer/manifests/keptn/charts/control-plane/templates/core.yaml (hunk 3 above)
Comment (bacherfl) [reply installer/manifests/keptn/charts/control-plane/templates/core.yaml:359]: the name should be provided in the values.yaml file, I just added the default to be safe

**Review by mowies — APPROVED** [review]

**Pull request merged**